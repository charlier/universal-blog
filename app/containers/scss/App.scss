body,
nav li,
nav ul {
    margin: 0;
    padding: 0;
}
body {
    background-color: #f4f4f4;
    font-family: 'Helvetica Neue', Arial, Helvetica, Verdana, sans-serif;
    color: #444;
}
nav {
    position: fixed;
    width:100%;
    top:0;
    left:0;
    box-shadow: 0 3px 6px rgba(0, 0, 0, .3);
    background-color: #3D9AD1;
    height: 70px;
    margin-bottom: 30px;
    overflow: visible;
}
nav li,
nav ul {
    display: block;
}
nav ul {
    overflow: hidden;
}
nav li {
    float: left;
}
nav a {
    width: 100px;
    height: 100px;
    background-color: #64A8D1;
    border-radius: 50%;
    display: table-cell;
    text-align: center;
    vertical-align: middle;
    color: #fff;
    transition: all .15s ease-in;
    transition-property: color, background-color;
    text-decoration: none;
}
nav a.active {
    text-decoration: underline;
}
nav a:hover {
    background-color: #9CF;
    color: #000;
}
#content {
    margin-top: 100px;
}
#loading {
  display: inline;
}
#loading:after {
  content:"";
  animation: load 2s linear infinite;
}
@keyframes load {
  0% {
    content: "";
  }
  25% {
    content: ".";
  }
  50% {
    content: "..";
  }
  75% {
    content: "...";
  }
}

#footer {
  font-size: 10px;
  border-top: 1px solid #eee;
  padding: 8px 0;
}
