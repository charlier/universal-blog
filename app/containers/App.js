import React, { Component } from 'react';
import { Link } from 'react-router';
import {IndexLink } from 'react-router';
import Helmet from 'react-helmet';
import './scss/App.scss';
import config from '../config';

class App extends Component {

  render() {
    return (
      <div>
      <Helmet
      title='loading...'
      titleTemplate='charlie.im - %s'
      meta={[
        {'char-set': 'utf-8'},
        {'http-equiv': 'x-ua-compatible', 'content': 'ie=edge'},
        {'name': 'viewport', 'content': 'width=device-width, initial-scale=1'},
        {'name': 'og:type', 'content': 'website'},
        {'name': 'og:site_name', 'content': 'charlie.im'},
        {'name': 'twitter:creator', 'content': '@charlierogers'},
        {'name': 'twitter:site', 'content': '@charlierogers'}
      ]}
      link={[
        {'rel': 'shortcut icon', 'href': `${config.staticHost}/favicon.ico`, 'type': 'image/x-icon'},
        config.env ? {'rel': 'stylesheet', 'href': `${config.staticHost}/style.min.css`} : {}
      ]}
      />
      <nav>
      <ul>
      <li><IndexLink to="/" activeClassName='active'>Home</IndexLink></li>
      <li><Link to='/about' activeClassName='active'>About</Link></li>
      <li><Link to='/'>Resum&eacute;</Link></li>
      </ul>
      </nav>
      <section id="content">
      {this.props.children}
      </section>
      <footer id="footer">
      Site scripting by Charles A Rogers. Copyright &copy; 2002-16.<br />
      Any resemblance between users and real persons living or dead is purely coincidental.<br />
      Do not fold or bend.
      </footer>
      </div>
    );
  }
}

App.propTypes = {
  children: React.PropTypes.object
};

export default App;
