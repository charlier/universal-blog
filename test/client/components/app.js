import { h } from 'preact';
import { shallow } from 'preact-render-spy';

import App from '../../../src/client/components/app';

describe('App Component', () => {
  it('Renders the App Component', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Adds the children content', () => {
    const wrapper = shallow(<App {...{ children: 'App content' }} />);
    expect(wrapper.find('div').find('div').text()).toBe('App content');
  });
});
