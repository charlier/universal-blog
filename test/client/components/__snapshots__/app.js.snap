// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Renders the App Component 1`] = `
preact-render-spy (1 nodes)
-------
<div>
  <h1>Header</h1>
  <ul>
    <li>
      <UnnamedComponent0 href="/">Home</UnnamedComponent0>
    </li>
    <li>
      <UnnamedComponent0 href="/wiki/Banana">Wiki: Banana</UnnamedComponent0>
    </li>
    <li>
      <UnnamedComponent0 href="/github/charlier">Github: charlier</UnnamedComponent0>
    </li>
    <li>
      <UnnamedComponent0 href="/about">About</UnnamedComponent0>
    </li>
  </ul>
  <div></div>
</div>

`;
